jbpm-persistence/jbpm-persistence-api/src/main/java/org/jbpm/persistence/api/integration/base/BaseEventCollection.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.persistence.api.integration.base;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import org.jbpm.persistence.api.integration.EventCollection;
+import org.jbpm.persistence.api.integration.InstanceView;
+
+/**
+ * Base event collection that collects all events and in LinkedHashSet to eliminate duplicates.
+ * No extra filtering is performed.
+ *
+ */
+public class BaseEventCollection implements EventCollection {
+
+    private static final long serialVersionUID = -5241582057875657702L;
+    private Set<InstanceView<?>> events = new LinkedHashSet<>();
+    
+    @Override
+    public void update(InstanceView<?> item) {
+        this.events.add(item);
+    }
+    
+    @Override
+    public void remove(InstanceView<?> item) {
+        this.events.add(item);
```

Comment:
fixed

Reply:
re-fixed it :) it was actually as it should. This collection records all operations though it will avoid  duplicates. The reason for it is if the process instance was started and finished with single transaction it will be removed - but in context of event manager if we would truly remove it then event manager wouldn't know that such instance was created and thus wouldn't record it at all. So the responsibility with what to do with such InstanceView is on the actual emitters some might store it some might discard instances that are already done.

Reply:
Understand, it is up to users to alter this as they wish. But at the first sight it was really weird :)